tests/selenium/test_sizing.py

```diff
@@ -0,0 +1,270 @@
+        row_deletable=True,
+    )
+
+    styles = [
+        dict(
+            style_table=dict(
+                width=500, minWidth=500, maxWidth=500, paddingBottom=10, display="none"
+            )
+        ),
+    ]
+
+    fixes = [
+        dict(),
+        dict(fixed_columns=dict(headers=True)),
+        dict(fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True), fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True, data=1)),
+        dict(fixed_rows=dict(headers=True, data=1)),
+        dict(
+            fixed_columns=dict(headers=True, data=1),
+            fixed_rows=dict(headers=True, data=1),
+        ),
+    ]
+
+    variations = []
+    i = 0
+    for style in styles:
+        for fix in fixes:
+            variations.append({**style, **fix, **base_props, "id": "table{}".format(i)})
+            i = i + 1
+
+    variations_range = range(0, len(variations))
+
+    tables = [DataTable(**variation) for i, variation in enumerate(variations)]
+
+    app = dash.Dash(__name__)
+    app.layout = Div(
+        children=[
+            Button(id="btn", children="Click me"),
+            DataTable(
+                **base_props,
+                id="table350",
+                style_table=dict(
+                    width=350, minWidth=350, maxWidth=350, paddingBottom=10
+                )
+            ),
+            DataTable(
+                **base_props,
+                id="table500",
+                style_table=dict(
+                    width=500, minWidth=500, maxWidth=500, paddingBottom=10
+                )
+            ),
+            DataTable(
+                **base_props,
+                id="table750",
+                style_table=dict(
+                    width=750, minWidth=750, maxWidth=750, paddingBottom=10
+                )
+            ),
+            Div(tables),
+        ]
+    )
+
+    @app.callback(
+        [Output("table{}".format(i), "style_table") for i in variations_range],
+        [Input("btn", "n_clicks")],
+        prevent_initial_call=True,
+    )
+    def update_styles(n_clicks):
+        if n_clicks == 1:
+            return [
+                dict(width=500, minWidth=500, maxWidth=500, paddingBottom=10)
+                for i in variations_range
+            ]
+        elif n_clicks == 2:
+            return [
+                dict(width=750, minWidth=750, maxWidth=750, paddingBottom=10)
+                for i in variations_range
+            ]
+        elif n_clicks == 3:
+            return [
+                dict(
+                    width=750,
+                    minWidth=750,
+                    maxWidth=750,
+                    display="none",
+                    paddingBottom=10,
+                )
+                for i in variations_range
+            ]
+        elif n_clicks:
+            return [
+                dict(width=350, minWidth=350, maxWidth=350, paddingBottom=10)
+                for i in variations_range
+            ]
```

Comment:
Change the table style to make them visible / hidden and/or resized

Reply:
Beautiful test :)
Looks like you started to collect all these `style_table` dicts up above in `styles` then changed tacks and put them inline down here... could probably be simplified a little, also we don't really need n clones, could just be `return [dict(...)] * variations_range`... but 🤷 not important for a test, the functionality is great!

Reply:
The test did change a lot between this version and the first implementation. Cleaned it up. https://github.com/plotly/dash-table/pull/793/commits/575aa2c4bc838474bcfa251c8596897be6c76d17